spec.md

```diff
@@ -122,18 +136,18 @@ If the manifest is not found in the registry, the response code MUST be `404 Not
 ##### Pulling Blobs
 
 To pull a blob, perform a `GET` request to a url in the following form:
-[2a](#Endpoints) `/v2/<name>/blobs/<digest>`
+[end-2](#endpoints) `/v2/<name>/blobs/<digest>`
 
 `<name>` is the namespace of the repository, and `<digest>` is the blob's digest.
 
-A GET request to an existing blob URL MUST provide the expected blob, with a reponse code that MUST be `200 OK`.
+A GET request to an existing blob URL MUST provide the expected blob, with a response code that MUST be `200 OK`.
 
 If the blob is not found in the registry, the response code MUST be `404 Not Found`.
 
 
 #### Push
-Pushing an image works in the opposite order as a pull: the blobs making up the layers are pushed first, and the
-manifest last.
+Pushing an artifact typically works in the opposite order as a pull: the blobs making up the artifact are uploaded first, and the
+manifest last. However, this order is not required, and you MAY upload content to the registry in any order.
```

Comment:
Agreed, the way this is phrased leads me to believe that a registry would be non-compliant if it disallowed a manifest PUT that references non-existing blobs. 

This is contradictory to the description of `BLOB_UNKNOWN`:

> This error MAY be returned when a blob is unknown to the registry in a specified repository. This can be returned with a standard get or if a manifest references an unknown layer during upload.

Reply:
@dmcgowan @jonjohnsonjr how is this wording:

```
Pushing an artifact typically works in the opposite order as a pull:
the blobs making up the artifact are uploaded first, and the manifest last.
Strictly speaking, content can be uploaded to the registry in any order,
but a registry MAY reject a manifest if it references blobs that are not
yet uploaded, resulting in a `BLOB_UNKNOWN` error (code-1).
```

Reply:
I've added this to most recent revision

Reply:
Looks good, but seems it should replace rather than add